com.unity.render-pipelines.universal/ShaderLibrary/UnityGBuffer.hlsl

```diff
@@ -114,6 +140,9 @@ FragmentOutput SurfaceDataToGbuffer(SurfaceData surfaceData, InputData inputData
     output.GBuffer1 = half4(surfaceData.specular.rgb, surfaceData.occlusion);            // specular        specular        specular        occlusion
     output.GBuffer2 = half4(packedNormalWS, surfaceData.smoothness);                     // encoded-normal  encoded-normal  encoded-normal  smoothness
     output.GBuffer3 = half4(globalIllumination, 1);                                      // GI              GI              GI              [optional: see OutputAlpha()] (lighting buffer)
+    #if _RENDER_PASS_ENABLED
+    output.GBuffer4 = inputData.positionCS.z;
+    #endif
```

Comment:
Seems too overboard to add an attachment for this and we can get that from depth texture?

Reply:
In this case this is a transient/memoryless attachment, so it's not very costly. To use depth texture we would require a depth prepass (as depth copy between GBuffer and Deferred passes would break renderpass making fb fetch impossible). Whereas depth attachment read is not supported on some platforms

Reply:
Yes, it is mandatory to have depth stored in a color texture with RenderPass. Depth attachment cannot be read back on iOS and for Vulkan we were told it is "faster" to have depth as color.